components/metamask/index.tsx

```diff
@@ -18,119 +36,178 @@ const Button = styled.button`
   const context = useWeb3React();
   const { library } = context;
 
-  const addHenkakuToken = async () => {
-    if (!library) { return }
+  const addToken = async () => {
+    if (!library) {
+      return;
+    }
     try {
       const wasAdded = await library.provider.request({
         method: "wallet_watchAsset",
         params: henkakuToken
       });
 
       if (wasAdded) {
-        alert('wow Epic! go back to discord / これで全てが終了だおめでとう')
+        alert("Wow Epic! Go back to discord / これで全てが終了だおめでとう");
       }
     } catch (e) {
-      console.log(e)
-      alert(`something went wrong ask admin for help (なんか変だな): ${e}`)
+      console.log(e);
+      alert(`Something went wrong ask admin for help (なんか変だな): ${e}`);
     }
-  }
+  };
 
   return (
-    <Button onClick={addHenkakuToken}>Add henkaku token / Henkakuトークンを追加する</Button>
-  )
-}
+    <Button onClick={addToken}>
+      Add HENKAKU token / HENKAKUトークンを追加する
+    </Button>
+  );
+};
 
-const AddPolygonNetwork: React.VFC =  () => {
+const AddPolygonNetwork: React.VFC = () => {
   const context = useWeb3React();
   const { library } = context;
 
-  const addPolygonNetwork = async() => {
-    if (!library) { return }
+  const addNetwork = async () => {
+    if (!library) {
+      return;
+    }
     try {
       await library.provider.request({
         method: "wallet_addEthereumChain",
         params: [polyGonNetwork]
       });
     } catch (e) {
-      console.log(e)
-      alert(`something went wrong ask admin for help (なんか変だな): ${e}`)
+      console.log(e);
+      alert(`Something went wrong ask admin for help (なんか変だな): ${e}`);
     }
-  }
+  };
   return (
-    <Button onClick={addPolygonNetwork}>Add Polygon Network / Polygonのネットワークに接続する</Button>
-  )
-}
+    <Button onClick={addNetwork}>
+      Add Polygon Network / Polygonのネットワークに接続する
+    </Button>
+  );
+};
 
 const ConnectWallet = () => {
-  const injectedConnector = new InjectedConnector({})
-  const { chainId, account, activate, active,library } = useWeb3React<Web3Provider>()
-  const onClick = () => {
-    activate(injectedConnector)
-  }
+  const injectedConnector = new InjectedConnector({});
+  const { chainId, account, activate, active, library } =
+    useWeb3React<Web3Provider>();
+
+  const [isMetaMask, setIsMetaMask] = useState(false);
+
+  const connectInjected = async () => {
+    try {
+      await activate(injectedConnector);
+    } catch (e) {
+      console.log(e);
+      alert(`Something went wrong ask admin for help (なんか変だな): ${e}`);
+    }
+  };
 
   useEffect(() => {
-    console.log(chainId, account, active)
-    },);
+    if (window.ethereum && window.ethereum.isMetaMask) {
+      setIsMetaMask(true);
+    } else {
+      console.log("Need to install MetaMask");
+    }
+  }, []);
 
   return (
     <>
-      {active ? '' : (
+      {isMetaMask ? (
+        ""
+      ) : (
+        <Helper>
+          <p>
+            MetaMask is not installed. For mobile, please use the MetaMask
```

Comment:
maybe this one is better ?
```suggestion
            MetaMask is not installed. For smartphone, use Metamask in-app browser instead of safarai and chrome in smartphone.
```

Reply:
Fixed in this commit. https://github.com/henkaku-center/shiniri/pull/6/commits/4e55062961926981db44ef96fca5e4e2d9254f01